ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -104,8 +104,59 @@ struct S3FileHandle {
 const double throughputTargetGbps = 5;
 const uint64_t partSize = 8 * 1024 * 1024; // 8 MB.
 
-// Setting SDK options.
-Aws::SDKOptions options;
+// Wrapper singleton class for initializing and terminating the AWS SDK.
```

Comment:
Perhaps explain that this class only initializes the AWS SDK once, and all other times you are just counting the number of times a reference to the AWS is being made. Same goes for terminate, and how you only call shutdown once at the end.

Reply:
I've moved the class to a separate file and added comments there.